slackhealthbot/services/withings/requests.py

```diff
@@ -1,54 +1,25 @@
-import functools
-
 import httpx
-from sqlalchemy.ext.asyncio import AsyncSession
+from authlib.integrations.starlette_client.apps import StarletteOAuth2App
 
 from slackhealthbot.database import models as db_models
 from slackhealthbot.services.withings import oauth
 
 
-def create_headers(oauth_access_token: str) -> dict[str, str]:
-    return {
-        "Authorization": f"Bearer {oauth_access_token}",
-    }
-
-
-def requires_oauth(func):
```

Comment:
Unfortunately it looks like we still need to have a wrapper like this for automatic token refresh on revoked token.

Authlib is only proactive with token refreshes if it detects the token is expired.

Reply:
Refreshing on an expired token is good enough, after all.